### PR TITLE
Filter RequiredBlockIndex on equipment notification.

### PR DESCRIPTION
**Pull request by Unengine**

### CODE REVIEW

**Review by ipdae — CHANGES_REQUESTED** [review]
test is broken

**Review by ipdae — APPROVED** [review]

File: .Lib9c.Tests/Model/ItemNotificationTest.cs [hunk 1]
```diff
@@ -143,9 +143,44 @@ public void EquipTwoRings()
...
+                .Where(r => r.ItemSubType == ItemSubType.Ring && r.Grade == 1);
+            foreach (var row in rows)
+            {
+                var guid = Guid.NewGuid();
+                var equipment = (Equipment)ItemFactory.CreateItemUsable(row, guid, 1, 0);
+                inventory.AddItem(equipment);
+            }
+
+            hasNotification = inventory.HasNotification(avatarState.level, 0);
+            Assert.False(hasNotification);
```
Comment (boscohyun) [inline .Lib9c.Tests/Model/ItemNotificationTest.cs:183]: Shouldn't it be'True' after a ring is added to your inventory?

Comment (Unengine) [reply .Lib9c.Tests/Model/ItemNotificationTest.cs:183]: `var equipment = (Equipment)ItemFactory.CreateItemUsable(row, guid, 1, 0);`
As you can see, its `RequiredIndex` is 1, so it should be false.

Comment (boscohyun) [reply .Lib9c.Tests/Model/ItemNotificationTest.cs:183]: Now I see.
How about making it easier to read using the `Theory` and `InlineData` attributes?
ex.
```
[Theory]
[InlineData(0, 0, true)]
[InlineData(0, 1, false)]
[InlineData(...)]
public void RequiredBlockIndexTest(long blockIndex, long requiredBlockIndex, bool hasNotification)
{ /* */ }
```

Comment (Unengine) [reply .Lib9c.Tests/Model/ItemNotificationTest.cs:183]: Seems like a good idea!

**Review by boscohyun — APPROVED** [review]

**Pull request merged**